File: examples-sdk/csharp/011-basic-graphql-api/Main.cs
// 011-basic-graphql-api — C# SDK equivalent
// Compile: vil compile --from csharp --input 011-basic-graphql-api/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("graphql-api", 8080);
var graphql = new ServiceProcess("graphql");
graphql.Endpoint("GET", "/", "index");
graphql.Endpoint("GET", "/schema", "schema_info");
graphql.Endpoint("GET", "/entities", "list_entities");
graphql.Endpoint("POST", "/query", "graphql_query");
server.Service(graphql);
server.Compile();

File: examples-sdk/java/011-basic-graphql-api/Main.java
// 011-basic-graphql-api — Java SDK equivalent
// Compile: vil compile --from java --input 011-basic-graphql-api/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilServer server = new VilServer("graphql-api", 8080);
        ServiceProcess graphql = new ServiceProcess("graphql");
        graphql.endpoint("GET", "/", "index");
        graphql.endpoint("GET", "/schema", "schema_info");
        graphql.endpoint("GET", "/entities", "list_entities");
        graphql.endpoint("POST", "/query", "graphql_query");
        server.service(graphql);
        server.compile();
    }
}
